Source/DafnyLanguageServer/Language/Symbols/DafnyLangSymbolResolver.cs

```diff
@@ -163,18 +201,41 @@ private class SymbolDeclarationResolver {
           cancellationToken.ThrowIfCancellationRequested();
           methodSymbol.Returns.Add(ProcessFormal(scope, result));
         }
-        ProcessAndRegisterMethodBody(methodSymbol, method.Body);
+        methodSymbol.Block = method.Body == null ? null : ProcessMethodBody(methodSymbol, method.Body);
```

Comment:
The `*Symbol` types were originally introduced for more flexibility when representing the symbol table. I can't remember the exact reason. But I think one was that I needed the possibility to dynamically adapt the locations of the symbols. I extracted the locations into  `SymbolTable.Locations` dictionary later since I didn't want to rebuild the `CompilationUnit`/Symbol tree with every change. This separation now indeed allows us to scrap the redundancy of the `*Symbol` types.

Reply:
Okay, so we could get rid of the `*Symbol` types to prevent having to extend them in future PRs, like this PR needed to do, right?

Reply:
Exactly. Moreover, since the API is purely internal, we don't have to adapt the integration tests when removing the `*Symbol` types.